scripts/tf_cnn_benchmarks/benchmark_cnn.py

```diff
@@ -1523,8 +1523,14 @@ def _build_image_processing(self, shift_ratio=0):
 
   def _build_model(self):
     """Build the TensorFlow graph."""
-    tf.set_random_seed(self.params.tf_random_seed)
-    np.random.seed(4321)
+    # Adjust seed so different workers start read different input files.
+    if self.params.variable_update == 'horovod':
+        import horovod.tensorflow as hvd
+        seed_adjustment = hvd.rank()
+    else:
+        seed_adjustment = 0
+    tf.set_random_seed(self.params.tf_random_seed + seed_adjustment)
+    np.random.seed(4321 + seed_adjustment)
```

Comment:
A better approach IMO would be to change the shift_ratio argument [here](https://github.com/tensorflow/benchmarks/blob/477c0d00674161a7c255819546c20908b4c094a6/scripts/tf_cnn_benchmarks/benchmark_cnn.py#L1641) to `hdv.rank()/hvd.size()`. If nonzero, shift_ratio is the fraction of batches that will be skipped initially. Setting to `hdv.rank()/hvd.size()` ensures that after an epoch's worth of images are processed, each image of the epoch was processed exactly once.

I also realized that in non-horovod distributed mode, shift_ratio is broken since we pass 0 instead of -1 in the link above. But that is a separate issue.